bun/components/TodoList.svelte

```diff
@@ -0,0 +1,123 @@
+<script>
+  // Define a Todo type using JSDoc for documentation purposes
+  /**
+   * @typedef {Object} Todo
+   * @property {string} text
+   * @property {boolean} completed
+   */
+
+  // Initialize an empty array for todos
+  /** @type {Todo[]} */
+  let todos = [];
+  let newTodo = "";
+
+  // Function to add a new todo
+  function addTodo() {
+    if (newTodo) {
+      todos = [...todos, { text: newTodo, completed: false }];
+      newTodo = "";
+    }
+  }
```

Comment:
```suggestion
   const reset = () => newTodo = "";

  const addTodo = () => {
    if (newTodo) {
      todos = [...todos, { text: newTodo, completed: false }];
      reset();
    }
  }
```
- move `function` declarations to `lambda` assignments
You'll see this used most often in our `es` modules. TL;DR - the `function` language feature enables `prototype` inheritance. My general approach to language features is: "don't use a language feature that gives you more than what you need unless you plan to use it". This can become dogmatic in practice, but the added benefit of 
```ts 
const myFunc = () => ""
```
 vs 
 ```ts
 function myFunc() {
  return "";
}
``` 
is that we're saving lines of code (LoC).

RE: moving `newTodo = ""` to `const reset = () => newTodo = ""` this is just a technique for code-level documentation.